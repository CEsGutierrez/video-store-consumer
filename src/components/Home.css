.landing-image-container {
  justify-content: center;
  display: flex;
}

.landing-image {
  height: 200px;
  width: auto;
  border-radius: 20px;
  filter: grayscale(50%);
}


.homeList {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.home {
  background-color: rgba(181, 181, 181, 50%);

  padding: 1em 0;
  margin: 0.5rem;

  min-height: 250px;
  min-width: 225px;
  flex-basis: 18%;

  border-radius: 5px;

  display: grid;
  grid-template-columns: 1fr 5fr 1fr;
  align-items: center;
}

.home__content {
  grid-column-start: 2;

  display: flex;
  flex-direction: column;
  justify-content: space-around;

  text-align: center;
  font-weight: 100;
  overflow: hidden;
}

.home__content-text {
  text-overflow: ellipsis;
  overflow: hidden;
  font-weight: bold;
}

.home__content-image {
  height: auto;
  width: 150px;
}

.home__content-text, .home__content-image {
  padding: 0;
  margin: 0;
}

.rentalButton {
  margin-top: 15px;
  align-self: center;
  font-family: 'Courier New', Courier, monospace;
  font-size: larger;
  width: 225px;
  height: 100px;
  border-radius: 50px;
  background: rgb(181, 181, 181, 50%);
  border: none;
}

.rentalButton:hover {
  cursor: pointer;
}
