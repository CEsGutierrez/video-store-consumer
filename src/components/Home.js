import React from 'react';
import PropTypes from 'prop-types';

const Home = ({createRentalCallback, resetMessageCallback}) => {
  
  return (
    <div>
      {resetMessageCallback()}
      <button
        // className="btn btn-primary pet-card--select-pet-btn"
        onClick={createRentalCallback}
        >
        Create a rental!
      </button>
    </div>
  )
}

Home.propTypes = {
  createRentalCallback: PropTypes.func.isRequired,
  resetMessageCallback: PropTypes.func.isRequired,
};

export default Home