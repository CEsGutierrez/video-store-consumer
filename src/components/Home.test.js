import React from 'react';
import { render, cleanup } from '@testing-library/react'
import Home from './Home';

describe('Home', () => {
  test('that it matches the existing snapshot', () => {
    // Arrange-Act
    const { asFragment } = render(
      <Home
      createRentalCallback={() => { }}
      resetMessageCallback={() => { }}
      />
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });

  test('The "createRentalCallback" function is called when the `Create a rental!` button is clicked on', () => {

    const createRental = jest.fn();

    const container = render(<Home
      createRentalCallback={createRental}
      resetMessageCallback={() => { }}
      />);

    const rentalButton = container.getByText(/Create a rental!/);
    rentalButton.click();

    expect(createRental).toHaveBeenCalled();
  });

});