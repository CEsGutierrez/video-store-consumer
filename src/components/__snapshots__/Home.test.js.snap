// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home that it matches the existing snapshot 1`] = `
<DocumentFragment>
  <div
    class="entire-page"
  >
    <button>
      Create a rental!
    </button>
  </div>
</DocumentFragment>
`;
